### ISSUE TITLE
Provide a super-set of package data from the foundryvtt.com website so that each Foundry VTT client can have a more broad understanding of modules which are not compatible with the current build.

### ISSUE THREAD

**Issue by Norc** [foundryvtt/foundryvtt#11130]
### User Experience

Currently, when a user goes to install packages in the Setup screen, packages that specify a maximum version that is older than their current Foundry version are silently not returned as valid search results.  There is no indication that the package exists at all.

While this makes sense in some cases, such as archived modules, it is causing significant user confusion when they cannot see their purchased Premium Content or their preferred game system. This has resulted in users reaching out for help on the Discord and sending customer support emails, particularly when a user is new to Foundry and has installed V12 as a fresh installation.

In cases where a Premium Content module or a Game System is not supported by the latest version of Foundry but is not archived, I would like the user to see a "stub" result with its name that explains the situation. Perhaps something like:

"This package is not compatible with the currently installed version of Foundry. You may want to consider _reverting to the previous version of Foundry_ until an update is available." The italicized section could be a link to a short KB article on that subject that I could create or a sentence of two of further instructions.

I'm not convinced that such a message would be necessary for an "ordinary" module that isn't premium content, but it may be a good idea for that also.

**Comment by aaclayton** [comment]
We can't do this in V12 but we can scope it for prioritization in V13.

**Comment by Norc** [comment]
Perhaps when we get to this in v13, for "important packages" (systems and premium content):
- 1 major verison out of date: clear warning template text available as a warning, but it is installable through the UI
- More than 1 major version out of date: Listed, but not installable.
Normal modules could follow the standard "it's up to date, or it doesn't show" method.

**Comment by unsoluble** [comment]
> but it is installable through the UI
Ehhhhh I don't think this should be altering the installable-or-not behaviour. If it's got a `compatibility.maximum`, that should still be honoured.

**Comment by JPMeehan** [comment]
> > but it is installable through the UI
> 
> Ehhhhh I don't think this should be altering the installable-or-not behaviour. If it's got a `compatibility.maximum`, that should still be honoured.

I think the point is "installable but can't be used" — so you could install SWADE but still couldn't try to open a world with it set as the system.

**Comment by unsoluble** [comment]
I don't understand why it should be installable if it can't be used. And if so, why grant this distinction to one generation over others?

**Comment by Norc** [comment]
I agree, good point Josh. Maximum compatible exceeded = uninstallable, full stop. There could be a different severity of text for a system / premium content item that is 1 version out of date and 2 major versions... or not. More of an optional aspect.

The most important thing is to have an immediate source of feedback to the user that they did search for the thing correctly, but thing is not up to date and what that means.

**Comment by unsoluble** [comment]
> an immediate source of feedback to the user that they did search for the thing correctly, but thing is not up to date and what that means

Yeah 100% agree here.

**Comment by KrazyCanucks** [comment]
> > I'm not convinced that such a message would be necessary for an "ordinary" module that isn't premium content, but it may be a good idea for that also.
> 
> I think systems usually being free content is a good argument for applying this change to all modules.
> 
> I've had at least one user ask why they can't see the system for the premium content they bought, I don't know how they got into that situation (for DCC, none of the free or premium content supports v12 yet) so I suspect it is an edge case.
> 
> If this change only applies to premium content this situation is likely to become more than just an edge case.

I believe I might have been that person.    

Recently in the last year I have bought a bunch of DCC books and wanted to run a DCC one-shot with some friends to get them into this system.   After seeing the sale on Foundry and on Foundry YouTube (marketing works), I decided to bite the bullet and buy a bunch of premium DCC VTT stuff because of the sale.   

I looked at both the Goodman Site and the DriveThru and eventually decided to by from DriveThru (Currency reasons).  Neither site indicated that the the premium VTT items only worked on V11. 

When buying through DriveThru it will automatically add the keys to your foundry account, which it did and I could see in my Foundry Profile.  I then opened Foundry and then could not find any of my bought content, and it did not show up in any of the searches.  I double-checked my Foundry Profile and rebooted my Foundry server etc and then started the whole google and discord search before going asking for help.  

I did not have the DCC system installed (I did have it before at one time)  and had upgraded to V12 for my 5e campaign.  Not seeing the DCC rule-set gave me an indication it might be a compatibility issue. 

I have been using foundry for over 2 years, but it just did not occur to go look at the modules on foundry BEFORE buying to determine the max level. And since this sale was done specifically in conjunction of the release of V12 and the yearly anniversary, there was every expectation that the content bought would work on V12


Working in IT, I always take LOTS of backups; so it was easy to revert to V11.  It was also happenstance that I was on a break from my 5e campaign and I had not run anything in V12 yet to worry about that data.

Also managing large IT systems at work, my users always say they would rather see data but have it display/read only as well,  It makes sense, and reduces SOO many questions. So to make a long story longer, I think having them seen but display only with a reason would be great and reduce so many questions for the content creators.

I appreciate all the work that goes into this system and the amount of work of the content creators. It is a passion of love that I see in all this work. 

Carry On, but make sure to search those tiles or have a 10' pole on you.
Scott
